libsoftwarecontainer/unit-test/gateway_unittest.cpp

```diff
@@ -186,16 +202,52 @@ TEST_F(GatewayTest, ConfigEnablesActivateEnablesTeardown) {
 }
 
 /*
- * Test that double activation is not possible
+ * Test that double activation is not possible for gatways that are
+ * not quick-launch enabled.
  */
-TEST_F(GatewayTest, CantActivateTwice) {
+TEST_F(GatewayTest, CanNotActivateTwiceWhenNotQuickLaunchEnabled) {
     loadConfig(validConf);
     ASSERT_TRUE(gw.setConfig(jsonConf));
     ASSERT_TRUE(gw.activate());
     ASSERT_THROW(gw.activate(), GatewayError);
 }
 
 /*
+ * Test that double activation is not possible for gatways that are
+ * quick-launch enabled.
+ */
+TEST_F(GatewayTest, CanNotActivateTwiceWhenIsQuickLaunchEnabled) {
+    loadConfig(validConf);
+    ASSERT_TRUE(quickLaunchGW.setConfig(jsonConf));
+    ASSERT_TRUE(quickLaunchGW.activate());
+    ASSERT_THROW(quickLaunchGW.activate(), GatewayError);
+}
+
+/*
+ * Test that double configuration and activation is not possible for
+ * gatways that are not quick-launch enabled.
+ */
+TEST_F(GatewayTest, CanNotConfigureAndActivateTwiceWhenNotQuickLaunchEnabled) {
+    loadConfig(validConf);
+    ASSERT_TRUE(gw.setConfig(jsonConf));
+    ASSERT_TRUE(gw.activate());
+    ASSERT_THROW(gw.setConfig(jsonConf), GatewayError);
+    ASSERT_THROW(gw.activate(), GatewayError);
```

Comment:
But the two tests do test two different behaviors. I had help by this test when working with the changes, so I thought it would be good to have to support future refactoring or changes in behavior. Just testing that activate fails does not define the behavior to the same extent that also testing activate fails even after a configure.